gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineVersions.java

```diff
@@ -53,19 +60,33 @@
 
     @Override
     public void apply(Project project) {
-        File rootVersionsPropsFile = rootVersionsPropsFile(project);
 
         // apply plugin: "nebula.dependency-recommender"
         project.getPluginManager().apply(DependencyRecommendationsPlugin.class);
 
         // get dependencyRecommendations extension
-        RecommendationProviderContainer extension = project.getExtensions()
+        RecommendationProviderContainer extension = project
+                .getExtensions()
                 .getByType(RecommendationProviderContainer.class);
 
-        extension.setStrategy(RecommendationStrategies.OverrideTransitives); // default is 'ConflictResolved'
+        extension.setStrategy(RecommendationStrategies.OverrideTransitives); // default is 'ConflictResolved';
+        extension.whenObjectAdded(recommendationProvider -> applyPropsFile(project));
 
-        extension.propertiesFile(ImmutableMap.of("file", rootVersionsPropsFile));
+        project.getTasks().create("checkBomConflict", BomConflictCheckTask.class);
+        project.getTasks().create("checkNoUnusedPin", NoUnusedPinCheckTask.class);
+        project.getPlugins().apply(BasePlugin.class);
+        project.getTasks().create("checkVersionsProps");
+        project.getTasks().getByName("checkVersionsProps").dependsOn("checkBomConflict", "checkNoUnusedPin");
```

Comment:
these two should happen as a configure action on `checkVersionProps`. You can do `project.getTasks().register(...)`